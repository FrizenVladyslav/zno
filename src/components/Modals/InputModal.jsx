import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Button, Modal, Input } from 'semantic-ui-react'

const InputModal = props => {
  const [inputValue, setInputValue] = useState('')

  return (
    <Modal size="mini" open={props.open} onClose={props.onClose}>
      <Modal.Header>{props.title}</Modal.Header>
      <Modal.Content>
        <Input
          defaultValue={props.defaultValue || ''}
          label={{ tag: !!props.label, content: props.label }}
          labelPosition="right"
          placeholder={props.placeholder}
          onChange={({ target: { value } }) => setInputValue(value)}
        />
      </Modal.Content>
      <Modal.Actions>
        <Button
          negative
          onClick={props.onClose}
          content="Cкасувати"
          icon="close"
        />
        <Button
          positive
          icon="checkmark"
          content="Підтвердити"
          onClick={() => props.onSuccess(inputValue)}
        />
      </Modal.Actions>
    </Modal>
  )
}

InputModal.propTypes = {
  defaultValue: PropTypes.string,
  label: PropTypes.string,
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onSuccess: PropTypes.func,
  placeholder: PropTypes.string,
  title: PropTypes.string,
}

export default InputModal
