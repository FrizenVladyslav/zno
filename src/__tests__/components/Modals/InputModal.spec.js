// @ts-nocheck
import React from 'react'
import { shallow } from 'enzyme'
import InputModal from 'components/Modals/InputModal'

describe('input modal', () => {
  const initProps = {
    onClose: jest.fn(),
    onSuccess: jest.fn(),
    open: true,
  }

  describe('render with props', () => {
    const initInputProps = {
      defaultValue: 'defaultValue',
      label: 'label',
      placeholder: 'placeholder',
    }
    const props = Object.assign(initProps, initInputProps, { title: 'title' })
    const inputModal = shallow(<InputModal {...props} />)

    it('renders properly', () => {
      expect(inputModal).toMatchSnapshot()
    })

    it('opened input modal correct rendered with props', () => {
      const expectedInputProps = {
        ...initInputProps,
        label: { tag: true, content: initInputProps.label },
      }
      const { label, placeholder, defaultValue } = inputModal
        .find('Input')
        .props()

      expect({ label, placeholder, defaultValue }).toEqual(expectedInputProps)
      expect(inputModal.find('ModalHeader').props().children).toBe(props.title)
    })
  })

  describe('when click the cancel button', () => {
    const inputModal = shallow(<InputModal {...initProps} />)

    beforeEach(() => {
      inputModal
        .find('Button')
        .first()
        .simulate('click')
    })

    it('call the props.onClose', () => {
      expect(initProps.onClose).toHaveBeenCalledTimes(1)
    })
  })

  describe('when click the success button', () => {
    const inputModal = shallow(<InputModal {...initProps} />)

    beforeEach(() => {
      inputModal
        .find('Button')
        .last()
        .simulate('click')
    })

    it('call the props.onSuccess', () => {
      expect(initProps.onSuccess).toBeCalledTimes(1)
    })
  })
})
