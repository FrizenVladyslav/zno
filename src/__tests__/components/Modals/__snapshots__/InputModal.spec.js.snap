// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input modal render with props renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputModal
    defaultValue="defaultValue"
    label="label"
    onClose={[MockFunction]}
    onSuccess={[MockFunction]}
    open={true}
    placeholder="placeholder"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "centered": true,
      "children": Array [
        <ModalHeader>
          title
        </ModalHeader>,
        <ModalContent>
          <Input
            defaultValue="defaultValue"
            label={
              Object {
                "content": "label",
                "tag": true,
              }
            }
            labelPosition="right"
            onChange={[Function]}
            placeholder="placeholder"
            type="text"
          />
        </ModalContent>,
        <ModalActions>
          <Button
            as="button"
            content="Cкасувати"
            icon="close"
            negative={true}
            onClick={[MockFunction]}
          />
          <Button
            as="button"
            content="Підтвердити"
            icon="checkmark"
            onClick={[Function]}
            positive={true}
          />
        </ModalActions>,
      ],
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "onClose": [MockFunction],
      "open": true,
      "size": "mini",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "title",
        },
        "ref": null,
        "rendered": "title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Input
            defaultValue="defaultValue"
            label={
              Object {
                "content": "label",
                "tag": true,
              }
            }
            labelPosition="right"
            onChange={[Function]}
            placeholder="placeholder"
            type="text"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultValue": "defaultValue",
            "label": Object {
              "content": "label",
              "tag": true,
            },
            "labelPosition": "right",
            "onChange": [Function],
            "placeholder": "placeholder",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Button
              as="button"
              content="Cкасувати"
              icon="close"
              negative={true}
              onClick={[MockFunction]}
            />,
            <Button
              as="button"
              content="Підтвердити"
              icon="checkmark"
              onClick={[Function]}
              positive={true}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "content": "Cкасувати",
              "icon": "close",
              "negative": true,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "content": "Підтвердити",
              "icon": "checkmark",
              "onClick": [Function],
              "positive": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "centered": true,
        "children": Array [
          <ModalHeader>
            title
          </ModalHeader>,
          <ModalContent>
            <Input
              defaultValue="defaultValue"
              label={
                Object {
                  "content": "label",
                  "tag": true,
                }
              }
              labelPosition="right"
              onChange={[Function]}
              placeholder="placeholder"
              type="text"
            />
          </ModalContent>,
          <ModalActions>
            <Button
              as="button"
              content="Cкасувати"
              icon="close"
              negative={true}
              onClick={[MockFunction]}
            />
            <Button
              as="button"
              content="Підтвердити"
              icon="checkmark"
              onClick={[Function]}
              positive={true}
            />
          </ModalActions>,
        ],
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [MockFunction],
        "open": true,
        "size": "mini",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "title",
          },
          "ref": null,
          "rendered": "title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              defaultValue="defaultValue"
              label={
                Object {
                  "content": "label",
                  "tag": true,
                }
              }
              labelPosition="right"
              onChange={[Function]}
              placeholder="placeholder"
              type="text"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultValue": "defaultValue",
              "label": Object {
                "content": "label",
                "tag": true,
              },
              "labelPosition": "right",
              "onChange": [Function],
              "placeholder": "placeholder",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Button
                as="button"
                content="Cкасувати"
                icon="close"
                negative={true}
                onClick={[MockFunction]}
              />,
              <Button
                as="button"
                content="Підтвердити"
                icon="checkmark"
                onClick={[Function]}
                positive={true}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "content": "Cкасувати",
                "icon": "close",
                "negative": true,
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "content": "Підтвердити",
                "icon": "checkmark",
                "onClick": [Function],
                "positive": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
